### ISSUE TITLE
2013 files are not parsing well!

### ISSUE THREAD

**Issue by laironald** [funginstitute/patentprocessor#16]
The 2013 files are not parsing well. :(
For citations-patents ratio for all parsed files in 2013, we see 0.20. This does not comparable to the historical ratio of 30. (This means, 2013 suggests that on average patents have 0.20 citations... weird number indeed!)

Here's a specific example: 
Looking up D680098, I see the following statistics. 
(I use alchemy and the test dataset, which includes 2013 information)

```
In [1]: from alchemy import *
In [2]: pat = session.query(Patent).filter(Patent.id == u"D680098").one()
In [3]: pat.stats()
Out[3]: 
{'assignees': 1,
 'citations': 0,
 'citedby': 0,  #ignore this now
 'classes': 3,
 'inventors': 0,
 'ipcrs': 0,
 'lawyers': 0,
 'otherreferences': 0,
 'relpatents': 0,  #ignore this now
 'usreldocs': 2}
```

This looks weird. No citations, inventors, lawyers, etc? Weird. Looking at the RAW XML (ipg130416.xml), I see the following below. So I know citations (I count 86 including otherreferences per patcit num count) and inventors exist (I see 4 total -- do you use applicant or inventors? are these two always consistent?). I also see info on lawyers.

``` <us-references-cited>
<us-citation>
<patcit num="00001">
<document-id>
<country>US</country>
<doc-number>D427169</doc-number>
<kind>S</kind>
<name>Saeki et al.</name>
<date>20000600</date>
</document-id>
</patcit>
<category>cited by applicant</category>
</us-citation>
...
</us-references-cited>

<inventor>
<inventor sequence="001" designation="us-only">
<addressbook>
<last-name>Fujii</last-name>
<first-name>Yoshito</first-name>
<address>
<city>Osaka</city>
<country>JP</country>
</address>
</addressbook>
</inventor>
...
</inventors>
```

**Comment by gtfierro** [comment]
I'll take a look at this today, but my initial guess is that it's the 

```
<inventor>
<inventor sequence="001" designation="us-only">
```

lines that are causing it to balk.

**Comment by laironald** [comment]
is this different than before?

On Friday, July 5, 2013, Gabe Fierro wrote:

> I'll take a look at this today, but my initial guess is that it's the
> 
> <inventor>
> <inventor sequence="001" designation="us-only">
> 
> lines that are causing it to balk.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20534800
> .

## 

Sent from mobile

**Comment by gtfierro** [comment]
I haven't seen that type of structure in an XML document before, but that doesn't mean it hasn't occurred. My guess is the two nested `<inventor>` tags are causing the parser to look inside the first one for the contents of what's really in the second, but I designed the parser to get around problems like that.

**Comment by laironald** [comment]
I see. they may have been depreciating the former tags for this new
structure

On Friday, July 5, 2013, Gabe Fierro wrote:

> I haven't seen that type of structure in an XML document before, but that
> doesn't mean it hasn't occurred. My guess is the two nested <inventor>tags are causing the parser to look inside the first one for the contents
> of what's really in the second, but I designed the parser to get around
> problems like that.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20534920
> .

## 

Sent from mobile

**Comment by gtfierro** [comment]
I'm running a 2013 parse now so I can isolate the error.

**Comment by laironald** [comment]
ok. thanks. when you get a chance, can you check abstracts. I can give you
more clarity on what is missing. when the data is loaded I can check this

On Friday, July 5, 2013, Gabe Fierro wrote:

> I'm running a 2013 parse now so I can isolate the error.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20535774
> .

## 

Sent from mobile

**Comment by Vadskye** [comment]
I independently ran into this error earlier today. I kept around a copy of the old parser, and it seems that the old parser also broke on 2013 files.

**Comment by laironald** [comment]
looks like none of the abstracts were parsed. I just checked the files. can
you check on this too?

On Friday, July 5, 2013, Gabe Fierro wrote:

> I'm running a 2013 parse now so I can isolate the error.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20535774
> .

## 

Sent from mobile

**Comment by gtfierro** [comment]
Yes, but can you file it as a separate issue?

**Comment by laironald** [comment]
yes sure. just submitted

On Friday, July 5, 2013, Gabe Fierro wrote:

> Yes, but can you file it as a separate issue?
> On Jul 5, 2013 1:13 PM, "Ron Lai" <notifications@github.com<javascript:_e({}, 'cvml', 'notifications@github.com');>>
> wrote:
> 
> > looks like none of the abstracts were parsed. I just checked the files.
> > can
> > you check on this too?
> > 
> > On Friday, July 5, 2013, Gabe Fierro wrote:
> > 
> > > I'm running a 2013 parse now so I can isolate the error.
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub<
> > 
> > https://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20535774>
> > 
> > > .
> > 
> > ## 
> > 
> > Sent from mobile
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20536983>
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20537039
> .

## 

Sent from mobile

**Comment by gtfierro** [comment]
This error is occurring because the tag names have changed, e.g. `<references-cited>` has become `<us-references-cited>`. There are a couple quick-fix solutions, but I think instead what I'll do is enable the parser to choose which xml parser it wants to use for a document. This will allow us to more flexibly deal with new schemas in the future.

**Comment by laironald** [comment]
gotcha. cool. thanks for poking into it.

On Fri, Jul 5, 2013 at 1:58 PM, Gabe Fierro notifications@github.comwrote:

> This error is occurring because the tag names have changed, e.g.
> <references-cited> has become <us-references-cited>. There are a couple
> quick-fix solutions, but I think instead what I'll do is enable the parser
> to choose which xml parser it wants to use for a document. This will allow
> us to more flexibly deal with new schemas in the future.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/funginstitute/patentprocessor/issues/16#issuecomment-20538667
> .

## 

sent from mobile

**Comment by gtfierro** [comment]
Fixed as of #20
